packages/opentelemetry-node/src/NodePluginManager.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { PluginManager } from '@opentelemetry/core';
+import { NodePluginManagerConfig } from './config';
+import { DEFAULT_INSTRUMENTATION_PLUGINS } from './config';
+import { PluginLoader, Plugins } from './instrumentation/PluginLoader';
+
+export class NodePluginManager extends PluginManager {
+  private readonly _pluginLoader: PluginLoader;
+
+  constructor(config: NodePluginManagerConfig = {}) {
+    super(config);
+
+    this._pluginLoader = new PluginLoader(
+      this.tracerProvider,
+      this.meterProvider,
+      this.logger
+    );
+
+    config.plugins
+      ? this._pluginLoader.load(
+          this._mergePlugins(DEFAULT_INSTRUMENTATION_PLUGINS, config.plugins)
+        )
+      : this._pluginLoader.load(DEFAULT_INSTRUMENTATION_PLUGINS);
+  }
+
+  stop() {
+    this._pluginLoader.unload();
+  }
+
+  /**
+   * Two layer merge.
+   * First, for user supplied config of plugin(s) that are loaded by default,
+   * merge the user supplied and default configs of said plugin(s).
+   * Then merge the results with the default plugins.
+   * @returns 2-layer deep merge of default and user supplied plugins.
+   */
+  private _mergePlugins(
+    defaultPlugins: Plugins,
+    userSuppliedPlugins: Plugins
+  ): Plugins {
+    const mergedUserSuppliedPlugins: Plugins = {};
+
+    for (const pluginName in userSuppliedPlugins) {
+      mergedUserSuppliedPlugins[pluginName] = {
+        // Any user-supplied non-default plugin should be enabled by default
+        ...(DEFAULT_INSTRUMENTATION_PLUGINS[pluginName] || { enabled: true }),
+        ...userSuppliedPlugins[pluginName],
+      };
+    }
+
+    const mergedPlugins: Plugins = {
```

Comment:
I don't remember exactly, but I think the deep merge implementation was not needed because we only have 2 levels of configs to worry about. I can't remember who implemented that who can confirm... maybe @naseemkullah?

Reply:
I did implement this, initially proposed we go with what appears to be a healthily maintained https://www.npmjs.com/package/deepmerge however due to there only being 2 levels of config too worry about we opted to write it up, seems like I screwed it up as per @obecny snippet above!